src/Config/DeserializationVariableReplacementSettings.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.RegularExpressions;
+using Azure.Core;
+using Azure.DataApiBuilder.Config.Converters;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Service.Exceptions;
+using Azure.Identity;
+using Azure.Security.KeyVault.Secrets;
+
+namespace Azure.DataApiBuilder.Config
+{
+    public class DeserializationVariableReplacementSettings
+    {
+        public bool DoReplaceEnvVar { get; set; } = true;
+        public bool DoReplaceAkvVar { get; set; } = true;
+        public EnvironmentVariableReplacementFailureMode EnvFailureMode { get; set; } = EnvironmentVariableReplacementFailureMode.Throw;
+
+        // @env\('  : match @env('
+        // @AKV\('  : match @AKV('
+        // .*?      : lazy match any character except newline 0 or more times
+        // (?='\))  : look ahead for ')' which will combine with our lazy match
+        //            ie: in @env('hello')goodbye') we match @env('hello')
+        // '\)      : consume the ') into the match (look ahead doesn't capture)
+        // This pattern lazy matches any string that starts with @env(' and ends with ')
+        // ie: fooBAR@env('hello-world')bash)FOO')  match: @env('hello-world')
+        // This matching pattern allows for the @env('<match>') to be safely nested
+        // within strings that contain ') after our match.
+        // ie: if the environment variable "Baz" has the value of "Bar"
+        // fooBarBaz: "('foo@env('Baz')Baz')" would parse into
+        // fooBarBaz: "('fooBarBaz')"
+        // Note that there is no escape character currently for ') to exist
+        // within the name of the environment variable, but that ') is not
+        // a valid environment variable name in certain shells.
+        public const string OUTER_ENV_PATTERN = @"@env\('.*?(?='\))'\)";
+        public const string OUTER_AKV_PATTERN = @"@AKV\('.*?(?='\))'\)";
+
+        // [^@env\(]   :  any substring that is not @env(
+        // [^@AKV\(]   :  any substring that is not @AKV(
+        // .*          :  any char except newline any number of times
+        // (?=\))      :  look ahead for end char of )
+        // This pattern greedy matches all characters that are not a part of @env()
+        // ie: @env('hello@env('goodbye')world') match: 'hello@env('goodbye')world'
```

Comment:
In the match here, the substring @env( is a substring but the pattern [^@env\(] expects any substring that is not @env(. so how did it match?